### PR TITLE
Adopt editor resolver for Welcome/Walkthrough pages

### PR DESCRIPTION
**Pull request by bhavyaus**
Addresses:
https://github.com/microsoft/vscode/issues/166831

### CODE REVIEW

File: src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.contribution.ts [hunk 1]
```diff
@@ -110,10 +110,13 @@ registerAction2(class extends Action2 {
 				}]);
 			} else if (!openedWalkthroughExists) {
 				// else open respecting toSide
-				editorService.openEditor(gettingStartedInput, { preserveFocus: toSide ?? false }, toSide ? SIDE_GROUP : undefined);
+				editorService.openEditor({
```
Comment (bpasero) [inline src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.contribution.ts:113]: There now seems to be a `const gettingStartedInput =` a few lines up that is not needed unless you are in the `if` branch above?

**Review by bpasero — CHANGES_REQUESTED** [review]
This is very cool!

I would suggest to tackle https://github.com/microsoft/vscode/issues/184673 here as well because depending on where the input is contributed, we may need to move it to an earlier phase. Specifically the editor resolver needs to know about the registration before we attempt to open the welcome page from `layout.ts` I think.

**Comment by bhavyaus** [comment]
> This is very cool!
> 
> I would suggest to tackle #184673 here as well because depending on where the input is contributed, we may need to move it to an earlier phase. Specifically the editor resolver needs to know about the registration before we attempt to open the welcome page from `layout.ts` I think.

I'll tackle this in separate PR. I might have to move some of startup logic to an earlier phase as well. I want to keep this change specific to adopting the editor resolver.

**Review by bpasero — APPROVED** [review]

**Comment by abhijit-chikane** [comment]
I think any tab open over there after closing should be able to get it after pressing CTRL+SHIFT+T 
I was going over the release notes and I accidently closed it but after click CTRL+SHIFT+T it didn't open
same for the extension tabs If I click on any extension and close the tab after CTRL+SHIFT+T it should open it 

I think any close tab should be undoable irrespective of the type of tab
@bhavyaus @bpasero

**Pull request merged**